BOJ/2828.사과_담기_게임/ajin.cpp

```diff
@@ -0,0 +1,42 @@
+#include <iostream>
+#include <vector>
+
+using namespace std;
+int M, N, J;
+int result;
+
+int main() {
+	ios_base::sync_with_stdio(false);
+	cin.tie(NULL);
+	cout.tie(NULL);
+
+	cin >> N >> M >> J;
+	vector<int> loc;
+	int tmp;
+	for (int i = 0; i < J; i++) {
+		cin >> tmp;
+		loc.push_back(tmp);
+	}
+
+	int l = 1;	int r = M;	// 바구니의 시작 칸(l)과 끝 칸(r) 위치
+	for (int i = 0; i < loc.size(); i++) {
+		int x = loc[i];	// 가야하는 위치
+		if (l <= x && x <= r) {
+			continue;
+		}
+		do {
+			if (r < x) {	// 가야하는 위치가 더 오른쪽에 있는 경우, 오른쪽으로 이동
+				l++; r++;
+			}
+			if (x < l) {	// 가야하는 위치가 더 왼쪽에 있는 경우, 왼쪽으로 이동
+				l--; r--;
+			}
+			result++;	// 1칸 이동
+		} while (l > x || x > r);	// l <= x <= r 를 만족하는 순간, 이동 종료
```

Comment:
위치 차이를 반복문이 아니어도 계산해서 예은님이 올려주신 코드처럼 움직여야 하는 위치를 계산할 수 있을 것 같아요!

시간 차이가 유의미하게 있을지는 모르겠네요 🤔


```suggestion
	
	if (r < x) {	// 가야하는 위치가 더 오른쪽에 있는 경우, 오른쪽으로 이동
		result += x - r;
        l += x - r;
        r = x;
    }
	else if (x < l) {	// 가야하는 위치가 더 왼쪽에 있는 경우, 왼쪽으로 이동
        result += l - x;
        r -= l - x;
        l = x;
	}
```